applications/wika_network.md

```diff
@@ -0,0 +1,446 @@
+| 0d. | Docker | We will provide a Dockerfile that packages the ETL service. |
+| 0e. | Article | We will publish a series of **articles** that explain: - The design of this repo and showcases typical queries that can be run against Elastic Search and Neo4J. - Fully reproducible guide to deploy Elastic Search along the ETL. - Fully reproducible guide to deploy Neo4J along the ETL. |
+| 1. | ETL | The ETL starts at the block number saved in the database and processes one block at a time. When done processing history, it will listen to new blocks and apply the same ETL. |
+| 2. | ETL Elastic Search |  Elastic Search will store one document per URL with title and other meta information pulled from the web page.    |  
+| 3. | ETL Neo4J | Neo4J will store one node per URL, one node per Account, Like relationships (edges) and Ownership relationships... |  
+| 4. | ETL Error Recovery | The service should report errors to its admin recipient by email and gracefully keep retrying a certain number of times before stopping.  | 
+| 5. | ETL Logging | Advanced monitoring is out of scope for this first version, but the service will write clean logs to report status. |  
+| 6. | Configuration | Database endpoints are defined in a configuration file. The configuration file location has a default, and can also be changed using en environment variable. The ETL also allows to do Elastic Search only, Neo4J only, or both.  |  
+| 7. | Example clients | Not part of the repo scope itself, but it will provide documentation on how to plug-in client tools. |  
+
+
+
+
+### Milestone 2 — Offchain API
+
+* **Estimated Duration:** 1 month
+* **FTE:**  1
+* **Costs:** 5,000 USD
+
+Global success criteria:
+* API is deployed on https://api.test.wika.network.
+* API is consumed by the UI in app.wika.network.
+
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 0a. | License | Apache 2.0 |
+| 0b. | Documentation | We will provide both **inline documentation** of the code, a **gituhb repo readme** that explains how to spin up the API endpoint, and also auto-generated API documentation in s standard format. |
+| 0c. | Testing Guide | Core functions will be fully covered by unit tests to ensure functionality and robustness. The **github  repo readme** we will also describe how to run these tests. |
+| 0d. | Docker | We will provide a Dockerfile that packages the API service. |
+| 0e. | Article | We will publish an **article** that explains the design of this repo and showcases the deployed functions.) |
+| 1. | list_url_by_liker(address) | List the URLs liked by input address. |
+| 2. | list_url_by_owner(address) |  List the URLs owned by input address.    |  
+| 3. | search(query) | Search URLs matching the query using domain, title and meta data |  
+| 4. | recommend(address) | Generate a list of recommendations for a user  | 
```

Comment:
How do you plan on generating recommendations? Do you store any other data than addresses and liked URLs?

Reply:
The chain only stores addresses and liked URLs, but as specified in the ETL, Elastic Search will store one document per URL with title and other meta information pulled from the web page. 
We plan on combining collaborative filtering (who liked same pages and what other pages they liked) combined with content based ML (with the data pulled by the ETL) to build a first version of the recommended function.
Please note that we really plan to make it the simplest thing that could possibly work as a proof of concept. Performance optimizations and scalability concerns will not be part of this initial scope.